flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/config/RateLimiterFactory.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.config;
+
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+
+import org.apache.flink.shaded.guava18.com.google.common.util.concurrent.RateLimiter;
+
+import java.util.Properties;
+
+/**
+ * A RateLimiterFactory that configures and creates a rate limiter.
+ */
+public class RateLimiterFactory {
+
+	/** Flag that indicates if ratelimiting is enabled. */
+	private static final String RATELIMITING_FLAG = "consumer.ratelimiting.enabled";
```

Comment:
Personally, I prefer independent configurations if possible. I am wondering if the following configuration names would be clear enough:
```
CONSUMER_MAX_BYTES_PER_SECOND_CONFIG="consumer.max.bytes.per.second"
CONSUMER_MAX_RECORDS_PER_SECOND_CONFIG="consumer.max.records.per.second"
```
In the future we can add configurations for the producers.

The default value of the above configurations could be either be set to Long.MAX_VALUE, or -1 to indicate not throttling.

Reply:
There is no need for this configuration. When a rate limiter is set on the consumer, then it is "enabled", otherwise it is null and no rate limiting takes place.

Reply:
Makes sense to enable ratelimiting based on whether or not a consumer's ratelimiter is null. To @becketqin's point, I think we would still need to pass the rate value itself from the consumer to its ratelimiter. Perhaps we could standardize that property across connectors?

Reply:
I ended up passing the rate value to the setter that sets the ratelimiter on the consumer. Let me know if that looks alright.